src/listContext.js

```diff
@@ -54,6 +54,11 @@ const ListContextProvider = props => {
     }
   };
 
+  //we use .set to add the time the item was purchased to an already existing document we search for the item.id
+  const addDatePurchased = (item, lastDatePurchased) => {
```

Comment:
This is nice and concise, with a single responsibility. Love it!

Reply:
this was tricky for me because i didn't know how to search for the item id so was stuck here for most of the time coding